caddyconfig/httpcaddyfile/directives.go

```diff
@@ -398,6 +398,21 @@ func sortRoutes(routes []ConfigValue) {
 		if len(jPM) > 0 {
 			jPathLen = len(jPM[0])
 		}
+
+		// if both directives have no path matcher, use whichever one
+		// has any kind of matcher defined first.
+		if iPathLen == 0 && jPathLen == 0 {
+			var iMatchers, jMatchers int
+			if len(iRoute.MatcherSetsRaw) > 0 {
+				iMatchers = 1
+			}
+			if len(jRoute.MatcherSetsRaw) > 0 {
+				jMatchers = 1
+			}
+			return iMatchers > jMatchers
```

Comment:
This is quite a bit shorter:

```suggestion
			return len(iRoute.MatcherSetsRaw) > 0 && len(jRoute.MatcherSetsRaw) == 0
```

Double-check my logic here :) But I think this is the same idea: i < j if i has a matcher and j doesn't.